block/manager.go

```diff
@@ -336,6 +395,71 @@ func (m *Manager) trySyncNextBlock(ctx context.Context, daHeight uint64) error {
 	return nil
 }
 
+// BlockStoreRetrieveLoop is responsible for retrieving blocks from the Block Store.
+func (m *Manager) BlockStoreRetrieveLoop(ctx context.Context) {
+	// waitCh is used to signal the block store retrieve loop, that it should check block store for new blocks
+	// blockStoreCond can be signalled in completely async manner, and goroutine below
+	// works as some kind of "buffer" for those signals
+	waitCh := make(chan interface{})
+	lastBlockStoreHeight := uint64(0)
+	go func() {
+		for {
+			m.blockStoreMtx.Lock()
+			m.blockStoreCond.Wait()
+			waitCh <- nil
+			m.blockStoreMtx.Unlock()
+			if ctx.Err() != nil {
+				return
+			}
+		}
+	}()
+	for {
```

Comment:
[nit] we should document infinite for loops so that it is clear what they are doing and what the exit conditions are.